syft/frameworks/torch/tensors/interpreters/polynomial.py

```diff
@@ -1,257 +1,355 @@
+            self.encrypt_fn["tanh"] = torch.tensor(self.tanh_coeffs)
+
+    def add_function(self, name, function, degree=10, min_val=-10, max_val=10, steps=100):
+        """Add function to function_attr dictionary.
+
+           Args:
+             name[str]: Name of function
+             degree[int]: Degree of function
+             piecewise[List]: List of piecewise functions in format [min_val of fit,max_val of fit,step of fit,function to fit values]
+             function[callable]: Base function
+           """
+
+        self.function_attr[name + "_degree"] = degree
+        self.function_attr[name + "_function"] = function
+        self.func_approx[name] = self.interpolate(
+            function, [min_val, max_val, steps], degree=degree
+        )
+
+    def interpolate(
+        self, function: Callable, interval: List[Union[int, float]], degree: int = 10
+    ) -> np.poly1d:
+
+        """Returns a interpolated version of given function using Numpy's polyfit method.
+        
+           Ref: https://mortendahl.github.io/2017/04/17/private-deep-learning-with-mpc/
+
+         Args:
+             function (a lambda function): Base function to be approximated
+             interval (list of floats/integers): Interval of values to be approximated
+             degree (Integer): Degree of polynomial approximation
+             precision (Integer): Precision of coefficients
+
+         Returns:
+             f_interpolated (Numpy poly1d): Approximated Function
+         """
+
+        # function we wish to approximate
+        f_real = function
+        # interval over which we wish to optimize
+        f_interval = np.linspace(interval[0], interval[1], interval[2])
+
+        # interpolate polynomial of given max degree
+        coefs = np.polyfit(f_interval, f_real(f_interval), degree)
+
+        # reduce precision of interpolated coefficients
+        precision = self.precision
+        coefs = [int(x * 10 ** precision) / 10 ** precision for x in coefs]
+
+        # approximation function
+        f_interpolated = np.poly1d(coefs)
+
+        return f_interpolated
+
+    def sigmoid(self, method="interpolation", degree=10, precision=10):
+        """Method provides Sigmoid function approximation interms of Taylor Series
+
+         Args:
+             x: Torch tensor
+
+         Returns:
+             approximation of the sigmoid function as a torch tensor
+         """
+
+        self.get_encrypt_function()
+        val = 0
+        x = self.child
+        if hasattr(self.encrypt_fn["sigmoid"], "child"):
+
+            for i in range(0, len(self.sigmoid_coeffs)):
+
+                val += (x ** i) * self.encrypt_fn["sigmoid"][
+                    (len(self.sigmoid_coeffs) - 1) - i
+                ].child
+
+        else:
+
+            for i in range(0, len(self.sigmoid_coeffs)):
+
+                val += (x ** i) * self.encrypt_fn["sigmoid"][(len(self.sigmoid_coeffs) - 1) - i]
+
+        return val
```

Comment:
This doesn't look like it's properly wrapped. 

On line 185: 
x = self.child

On line 190:
val += (x ** i) + .....

which means val is the same type as x

Thus, val is the same type as self.child which means it's the wrong type. 

Relatedly, shouldn't there be a hooking decorator on this method?

Reply:
It works fine now :)